pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -2112,14 +2113,26 @@ func (og *operationGenerator) checkIfSupportsNodeExpansion(volumeToMount VolumeT
 		return false, nil
 	}
 
+	// Get volume plugin to check if the corresponding CSI driver supports NodeExpand
+	volumePlugin, _ := og.volumePluginMgr.FindPluginBySpec(volumeToMount.VolumeSpec)
+	if volumePlugin.GetPluginName() == csi.CSIPluginName {
+		csiNodeExpandSupported, _ := csi.DoesPluginSupportNodeExpand(volumeToMount.VolumeSpec)
+		if !csiNodeExpandSupported {
+			return false, nil
+		}
+	}
+
 	// Get expander, if possible
 	expandableVolumePlugin, _ :=
 		og.volumePluginMgr.FindNodeExpandablePluginBySpec(volumeToMount.VolumeSpec)
+
+	// Check if node expansion is supported
 	if expandableVolumePlugin != nil &&
 		expandableVolumePlugin.RequiresFSResize() &&
```

Comment:
I considered that approach, but it would require a larger refactor of all `RequiresFSResize` calls to pass in `spec *volume.Spec`. I am okay doing that change, but I wasn't sure if that was the right approach.